File: multi_agent/projects/im-mobile/lib/services/translation_service.dart
import 'dart:convert';
import 'package:http/http.dart' as http;
import '../models/message_translation.dart';

class TranslationService {
  static const String _baseUrl = '/api/translation';
  final String _token;

  TranslationService(this._token);

  Map<String, String> get _headers => {
    'Authorization': 'Bearer $_token',
    'Content-Type': 'application/json',
  };

  Future<MessageTranslation> translate({
    required int messageId,
    required int userId,
    required String text,
    required String targetLang,
    String? sourceLang,
    bool autoTranslate = false,
  }) async {
    final response = await http.post(
      Uri.parse('$_baseUrl/translate'),
      headers: _headers,
      body: jsonEncode({
        'messageId': messageId,
        'userId': userId,
        'text': text,
        'targetLang': targetLang,
        'sourceLang': sourceLang,
        'autoTranslate': autoTranslate,
      }),
    );
    final data = jsonDecode(utf8.decode(response.bodyBytes));
    return MessageTranslation.fromJson(data['data']);
  }

  Future<List<MessageTranslation>> batchTranslate(List<Map<String, dynamic>> requests) async {
    final response = await http.post(
      Uri.parse('$_baseUrl/translate/batch'),
      headers: _headers,
      body: jsonEncode(requests),
    );
    final data = jsonDecode(utf8.decode(response.bodyBytes));
    return (data['data'] as List).map((e) => MessageTranslation.fromJson(e)).toList();
  }

  Future<TranslationSettings> getSettings(int userId) async {
    final response = await http.get(
      Uri.parse('$_baseUrl/settings').replace(queryParameters: {'userId': userId.toString()}),
      headers: _headers,
    );
    final data = jsonDecode(utf8.decode(response.bodyBytes));
    return TranslationSettings.fromJson(data['data']);
  }

  Future<TranslationSettings> updateSettings(int userId, TranslationSettings settings) async {
    final response = await http.put(
      Uri.parse('$_baseUrl/settings').replace(queryParameters: {'userId': userId.toString()}),
      headers: _headers,
      body: jsonEncode(settings.toJson()),
    );
    final data = jsonDecode(utf8.decode(response.bodyBytes));
    return TranslationSettings.fromJson(data['data']);
  }
}

File: im-desktop/src/services/translation-service.ts
import axios from 'axios';
import { TranslationRequest, TranslationSettings, MessageTranslation } from '../types/message-translation';
import { useAuthStore } from '../stores/auth-store';

const API_BASE = '/api/translation';

function getHeaders() {
  const token = useAuthStore.getState().token;
  return { Authorization: `Bearer ${token}` };
}

/**
 * 翻译文本
 */
export async function translate(request: TranslationRequest): Promise<MessageTranslation> {
  const response = await axios.post(`${API_BASE}/translate`, request, { headers: getHeaders() });
  return response.data.data;
}

/**
 * 批量翻译
 */
export async function batchTranslate(requests: TranslationRequest[]): Promise<MessageTranslation[]> {
  const response = await axios.post(`${API_BASE}/translate/batch`, requests, { headers: getHeaders() });
  return response.data.data;
}

/**
 * 获取翻译设置
 */
export async function getSettings(userId: number): Promise<TranslationSettings> {
  const response = await axios.get(`${API_BASE}/settings`, {
    params: { userId },
    headers: getHeaders()
  });
  return response.data.data;
}

/**
 * 更新翻译设置
 */
export async function updateSettings(userId: number, settings: TranslationSettings): Promise<TranslationSettings> {
  const response = await axios.put(`${API_BASE}/settings`, settings, {
    params: { userId },
    headers: getHeaders()
  });
  return response.data.data;
}
